service/history/queueProcessor.go

```diff
@@ -279,7 +279,14 @@ func (p *queueProcessorBase) processTaskAndAck(notificationChan <-chan struct{},
 		scope, err = p.processTaskOnce(notificationChan, task, logger)
 		return p.handleTaskError(scope, startTime, notificationChan, err, logger)
 	}
-	retryCondition := func(err error) bool { return true }
+	retryCondition := func(err error) bool {
+		select {
+		case <-p.shutdownCh:
```

Comment:
I thought we retry for a maximum of 10 seconds.  How can this cause go-routine leak? The pump retrying tasks should check for shutdownCh and bail out of the pump.

Reply:
the retry can possibly cause blocking on standby task, so, should check whether to continue to do an retry